syntax = "proto3";
package execute;




import "model/tool/callback.proto";
import "google/protobuf/struct.proto";

/*
ExecuteTool请求
*/
message ExecuteToolRequest {
message Metadata {
    /*
    页面访问地址
    */
    string origin = 1 ;
}
        
    
    /*
    执行时输入参数
  "inputs": {
    "@agents": [
    {
      "ip": "192.168.100.163",
      "instanceId": "5caee6c67d86a"
    }
    ],
    "ddddd": "5caee6c67d86a",
    "test1": "test params",
    "test2": "develop_1"
  },

    */
    google.protobuf.Struct inputs = 1 ;
    /*
    执行工具的agents列表
    */
    repeated string agents = 2 ;
    /*
    工具执行成功的回调信息
    */
    tool.Callback callback = 3 ;
    /*
    工具执行的通知信息
    */
    string needNotify = 4 ;
    /*
    工具执行通知的用户
    */
    string name = 5 ;
    /*
    一些拓展字段 目前支持origin=页面访问地址
    */
    Metadata metadata = 6 ;
    /*
    工具Id，服务端自动生成
    */
    string toolId = 7 ;
    /*
    工具版本
    */
    string vId = 8 ;
    /*
    执行用户
    */
    string execUser = 9 ;
    /*
    工具执行的超时时间(0表示不超时，默认为86400秒)
    */
    int32 timeout = 10 ;
}

/*
ExecuteTool返回
*/
message ExecuteToolResponse {
    /*
    工具执行id
    */
    string execId = 1 ;
}

/*
ExecuteToolApi返回
*/
message ExecuteToolResponseWrapper {
    /*
    返回码
    */
    int32 code = 1 ;
    /*
    返回码解释
    */
    string codeExplain = 2 ;
    /*
    错误详情
    */
    string error = 3 ;
    /*
    返回数据
    */
    execute.ExecuteToolResponse data = 4 ;
}

